projects/kit/components/range/range-change.directive.ts

```diff
@@ -0,0 +1,102 @@
+import {DOCUMENT} from '@angular/common';
+import {Directive, ElementRef, Inject} from '@angular/core';
+import {clamp, round, typedFromEvent} from '@taiga-ui/cdk';
+import {TUI_FLOATING_PRECISION} from '@taiga-ui/kit/constants';
+import {merge} from 'rxjs';
+import {filter, map, repeat, startWith, switchMap, takeUntil, tap} from 'rxjs/operators';
+
+import {TuiRangeComponent} from './range.component';
+
+// @dynamic
+@Directive({
+    selector: 'tui-range',
+})
+export class TuiRangeChangeDirective {
+    /**
+     * TODO replace with pointer events (when all supported browsers can handle them).
+     * Dont forget to use setPointerCapture instead of listening all documentRef events
+     */
+    private readonly pointerDown$ = merge(
+        typedFromEvent(this.elementRef.nativeElement, 'touchstart', {passive: true}).pipe(
+            filter(({touches}) => touches.length === 1),
+            map(({touches}) => touches[0]),
+        ),
+        typedFromEvent(this.elementRef.nativeElement, 'mousedown', {passive: true}),
+    );
+
+    private readonly pointerMove$ = merge(
+        typedFromEvent(this.documentRef, 'touchmove').pipe(
+            filter(({touches}) => touches.length === 1),
+            map(({touches}) => touches[0]),
+        ),
+        typedFromEvent(this.documentRef, 'mousemove'),
+    );
+
+    private readonly pointerUp$ = merge(
+        typedFromEvent(this.documentRef, 'touchend', {passive: true}),
+        typedFromEvent(this.documentRef, 'mouseup', {passive: true}),
+    );
+
+    constructor(
+        @Inject(DOCUMENT) private readonly documentRef: Document,
+        @Inject(ElementRef) private readonly elementRef: ElementRef<HTMLElement>,
+        @Inject(TuiRangeComponent) private readonly range: TuiRangeComponent,
+    ) {
+        let activeThumb: 'left' | 'right';
+
+        this.pointerDown$
+            .pipe(
+                tap(({clientX, target}) => {
+                    activeThumb = this.detectActiveThumb(clientX, target);
+                    elementRef.nativeElement.focus();
```

Comment:
I wonder if we should just drop `setNativeFocused` completely and use native method everywhere, like here 🤔

Reply:
I removed usage of `setNativeFocused` from `kit/components/range`-directory